plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Transformer } from './index';
+
+// requires repo
+export const addGitFeedbackLink = (): Transformer => {
+  return dom => {
+    const contentContainer = dom.querySelector(
+      '.md-content__inner',
+    ) as HTMLElement;
+    const sourceAnchor = dom.querySelector(
+      '.md-content__inner>a',
+    ) as HTMLAnchorElement;
+    // don't show if Gitlab or Github hosting not specified
+    if (!sourceAnchor || !sourceAnchor.href) {
+      return dom;
+    }
+    let gitHost = '';
+
+    if (sourceAnchor.href.includes('gitlab')) {
+      gitHost = 'gitlab';
+    } else if (sourceAnchor.href.includes('github')) {
```

Comment:
We should highlight that this will not work for ghe or any self-hosted GitHub/GitLab where the host does not contain the string `gitlab` or `github`. e.g. `ghe.mycompany.net` or `git.mycompany.com`.

Reply:
You could use `giturlparse` to detect self hosted instances.

Reply:
Or the @backstage/integrations package

Reply:
I'm stumped. For URLs that don't include the name of the implementation as a string, not sure if we can discern without additional information. `giturlparse` doesn't specify the provider, and the docs may be built externally and hence use a Git hosting solution that isn't set up in Backstage's integrations config.

Reply:
I've tried to add keys in app-config at techdocs.feedback.gitlab and .github for matching as Gitlab/Github Git hosting types, but having some problems making values added there appear in the config object. Perhaps there is some schema or config getter i need to modify that I'm unable to find so far?

Reply:
Fixed above. There are now optional keys in app-config under techdocs.feedback.gitlab and .github for specifying hostnames that should be identified as Gitlab or Github hosting.